File: 1557-Minimum-Number-of-Vertices-to-Reach-all-Nodes/solution.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        List<Integer> ans = new ArrayList<Integer>();
        Set<Integer> endset = new HashSet<Integer>();

        for (List<Integer> edge: edges) {
            endset.add(edge.get(1));
        }

        for (int i = 0; i < n; ++i) {
            if (!endset.contains(i)) {
                ans.add(i);
            }
        }
        return ans;
    }
}
// 

File: 1557-Minimum-Number-of-Vertices-to-Reach-all-Nodes/solution.cpp
class Solution {
public:
     vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> res, seen(n);
        for (auto& e: edges)
            seen[e[1]] = 1;
        for (int i = 0; i < n; ++i)
            if (seen[i] == 0)
                res.push_back(i);
        return res;
    }
};